core/src/main/scala/kafka/server/KafkaServerStartable.scala

```diff
@@ -17,12 +17,20 @@
 
 package kafka.server
 
-import kafka.utils.Logging
+import java.util.Properties
+
+import kafka.metrics.KafkaMetricsReporter
+import kafka.utils.{VerifiableProperties, Logging}
 
 
 class KafkaServerStartable(val serverConfig: KafkaConfig) extends Logging {
   private val server = new KafkaServer(serverConfig)
 
+  def this(serverProps: Properties) {
+    this(KafkaConfig.fromProps(serverProps))
+    KafkaMetricsReporter.startReporters(new VerifiableProperties(serverProps))
+  }
```

Comment:
IMO, this would be better as a method in the companion object:

`KafkaServerStartable.fromProps` or `KafkaServerStartable.apply`.

Secondary constructors are weird in Scala (the scoping is not quite right) and are generally avoided. Other advantages of the companion object is that you don't need to use `new` and you can start the reporters before creating the instance (which is the previous behaviour).

An alternative would be to have a method in the companion object that took a KafkaConfig and started the reporters. Then callers could pass properties by using `KafkaConfig.fromProps`. It is a bit weird that the behaviour with regards to the reporters differs solely on whether one passes a `KafkaConfig` or `Properties`. After all, these are equivalent in terms of the data they provide.